dexlib2/src/main/java/org/jf/dexlib2/HiddenApiRestriction.java

```diff
@@ -105,10 +107,10 @@ public boolean isDomainSpecificApiFlag() {
         HiddenApiRestriction normalRestriction = hiddenApiFlags[value & HIDDENAPI_FLAG_MASK];
 
         int domainSpecificPart = (value & ~HIDDENAPI_FLAG_MASK);
-        if (domainSpecificPart == 0) {
+        if (domainSpecificPart == 0 | domainSpecificPart != value) {
             return ImmutableSet.of(normalRestriction);
         }
-        return ImmutableSet.of(normalRestriction, domainSpecificApiFlags[(domainSpecificPart >> 3) - 1]);
+        return ImmutableSet.of(normalRestriction, domainSpecificApiFlags[(domainSpecificPart >> 4) - 1]);
```

Comment:
The normal api flags still use the first 3 bits, with the domain specific flags starting on the 4th bit. Shifting by 4 will shift off the first bit of the domain specific part.